### PR TITLE
K8s version bump

### PR DESCRIPTION
**Pull request by x13n**
Dependency update to work with k8s 1.14.

**What this PR does**
This migrates from deprecated godep to dep for managing dependencies, bumps versions of these dependencies and removes some dead code (although not all of it). My goal here was to update dependencies, so I removed only the parts that were failing tests or builds AND were unused. By unused I mean everything that is not a part of exporting data from kubelet Summary API into Stackdriver. Overview of changes:
- Riemann and Elasticsearch sinks were removed.
- Metrics API serving was removed.
- Integration tests were removed (they were testing only the Metrics API).
- Migrated from glog to klog.
- Switched dependency tracking from godep to dep.

PTAL @bmoyles0117 @kawych @wangzhen127

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,9 +56,6 @@ endif
 test-unit-cov: clean sanitize build
 	hooks/coverage.sh
 
-test-integration: clean build
```
Comment (kawych) [inline Makefile:64]: Why is it OK to remove integration tests? Do they test only the metrics API?

File: common/elasticsearch/OWNERS [hunk 2]
```diff
@@ -1,8 +0,0 @@
-reviewers:
```
Comment (kawych) [inline common/elasticsearch/OWNERS:1]: Is common/elasticsearch entirely for the purpose of elasticsearch sink? Are you removing the whole elasticsearch sink?

**Comment by kawych** [comment]
Mostly LGTM. Please:
- List all the changes somehow (either in the description or in the commit messages). I'd like to make sure I didn't overlook something important and that all of the changes are intended. Feel free to skip the most minor ones (like adjusting the code to some updated struct), but changes such as deleting whole sinks should be definitely documented.
- In the future, please split up any code changes from vendor changes. This is not fun to review.

File: Makefile (hunk 1 above)
Comment (x13n) [reply Makefile:64]: Yes.

File: common/elasticsearch/OWNERS (hunk 2 above)
Comment (x13n) [reply common/elasticsearch/OWNERS:1]: Yes.

**Comment by x13n** [comment]
I've added a high-level list of changes in the PR description above.

**Review by kawych — APPROVED** [review]

**Pull request merged**